Neos.Flow/Classes/Mvc/Routing/Dto/RouteTags.php

```diff
@@ -0,0 +1,145 @@
+<?php
+namespace Neos\Flow\Mvc\Routing\Dto;
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations as Flow;
+
+/**
+ * RouteTags to be associated with matched/resolved routes
+ *
+ * RouteTags can be added by Route Part handlers via the ResolveResult/MatchResult return values
+ * The tags will be added to the corresponding cache entries, so that they can be flushed selectively using the RouterCachingService
+ *
+ * @Flow\Proxy(false)
+ */
+final class RouteTags
+{
+
+    /**
+     * Pattern a tag must match. @see \Neos\Cache\Frontend\FrontendInterface::PATTERN_TAG
+     */
+    const PATTERN_TAG = '/^[a-zA-Z0-9_%\-&]{1,250}$/';
+
+    /**
+     * @var string[] numeric array of strings satisfying the PATTERN_TAG regex
+     */
+    private $tags = [];
+
+    /**
+     * @param string[] $tags numeric array of strings satisfying the PATTERN_TAG regex
+     */
+    private function __construct(array $tags)
+    {
+        $this->tags = $tags;
+    }
+
+    /**
+     * Creates an instance without any tags
+     *
+     * @return RouteTags
+     */
+    public static function createEmpty(): self
+    {
+        return new static([]);
+    }
+
+    /**
+     * Creates an instance with one given tag
+     *
+     * @param string $tag Tag value satisfying the PATTERN_TAG regex
+     * @return RouteTags
+     */
+    public static function createFromTag(string $tag): self
```

Comment:
createFromString ? Sounds strange Tag::createFromTag ?

Reply:
Indeed, `createFromString()` seems way better.

Reply:
A tag is a string in our case, so I think it is more explicit actually: Create a new tag collection from a single tag.
`createFromString` would mean: Create a tag collection from an arbitrary string wich would IMO suggest that you could do sth like `RouteTags::createFromString("tag1, "tag2")` which you can't right now.. But, true, I could change that

Reply:
I'm with Bastian here, `createFromString` kind of implies that it generates a collection with possibly multiple entries from a single (or multiple, but then I'd rather call it `createFromStrings`) string. The method is supposed to just convert a single Tag (guess we just really need a VO/TypeClass for that) into a collection.

Reply:
Reading this now I would actually say a Tag should probably be it's own object created from a string. :D

Reply:
@kitsunet yes, that's what I suggested with https://github.com/neos/flow-development-collection/pull/1126#discussion_r156880274 - I'll take care of kickstarting that to make up for todays fuckups ;)